### ISSUE TITLE
SDK version 4.1.0  exits immediately after launch

### ISSUE THREAD

**Issue by tadfisher** [plaid/plaid-link-android#260]
# The problem

Using the native SDK, launching via the `OpenPlaidLink()` contract results in a short loading screen, then LinkActivity finishes immediately after calling the `workflow` endpoint. The Link flow works correctly on version 4.0.0, and exhibits this behavior on version 4.1.0.

When this occurs, no logs are printed to Logcat (with `LinkLogLevel.VERBOSE`) and we receive a `LinkExit result` with all fields set to `null`. No activity is logged with my `client_user_id` in either Logs or Item Debugger.

# Environment

|                        |  |
| ------------------------| ---------- |
| Android OS Version                    | 14 (34)                |
| Android Devices/Emulators    | Pixel 7 Pro, Pixel 6 Pro, emulator                |

# Steps to Reproduce

Token configuration:

```
{
   "client_id":"OMITTED",
   "secret":"OMITTED",
   "user":{
      "client_user_id":"OMITTED",
   },
   "client_name":"Mercury",
   "products":[
      "auth",
      "transactions"
   ],
   "country_codes":[
      "US"
   ],
   "language":"en",
   "webhoook":"OMITTED",
   "android_package_name":"com.mercury.bank.dev"
}
```

# Logs

[logcat session](https://github.com/plaid/plaid-link-android/files/13956879/logcat.txt)

# Code To Reproduce Issue

This is a Compose application, but that shouldn't affect much (and this code works with 4.0.0):

```kotlin
// For example
@Composable
fun LaunchPlaidButton(val token: String, onResult: (LinkResult) -> Unit) {
    val launcher = rememberLauncherForActivityResult(OpenPlaidLink(), onResult)
    Button(
        onClick = {
            launcher.launch(linkTokenConfiguration {
                logLevel = LinkLogLevel.VERBOSE
                this.token = token
            })
        },
    ) {
        Text("Launch Plaid Link")
    }
}
```

**Comment by dseverns-livefront** [comment]
Plus one to this, just started working with Plaid so no example of working but exactly the same scenario listed here. Also attempted with this sample app and same result.

**Comment by eric-labelle** [comment]
@tadfisher / @dseverns-livefront mind looking with StrictMode? Probably [the same issue](https://github.com/plaid/plaid-link-android/issues/259) I filed earlier this week.

**Comment by melissaosullivan** [comment]
Hi, 

I double checked that this sample app is not seeing this auto-close issue on 4.1.0. I also built a small demo with compose and I can't reproduce the issue described. Could you please provide a sample app which displays this issue?

Thanks,
Melissa

**Comment by dseverns-livefront** [comment]
@melissaosullivan I tried this with the Plaid sample, only change I made was changing the product to `auth` to match my Plaid accounts, Can confirm after I tried version 4.0.0 it worked for me.

**Comment by dseverns-livefront** [comment]
@eric-labelle 4.1 was not working for me with and without strict mode enabled and never saw that stack trace. 4.0 works for both configs though.

**Comment by melissaosullivan** [comment]
@dseverns-livefront I'm not able to reproduce this issue so far. A few questions:
- What do you mean by "changing the product to `auth` to match my Plaid accounts?" Is this referring to the token creation request?
- Are you using a sandbox token?

**Comment by melissaosullivan** [comment]
@tadfisher @dseverns-livefront I think I've found the issue and addressed this in the [4.1.1 release](https://github.com/plaid/plaid-link-android/releases/tag/v4.1.1). Please try out this version. Thanks.

**Comment by dseverns-livefront** [comment]
>What do you mean by "changing the product to auth to match my Plaid accounts?" Is this referring to the token creation request?
Are you using a sandbox token?

Sorry, I changed the array in the server config to replace "transactions" with "auth" and yes it was sandbox tokens

**Comment by melissaosullivan** [comment]
Closing this ticket as this issue has been addressed in 4.1.1. Please upgrade to that version. Thanks for reporting this bug!
